src/commands/cmd_query.c

```diff
@@ -139,6 +140,8 @@ void _MGraph_Query(void *args) {
 cleanup:
 	// Release the read-write lock
 	if(lockAcquired) {
+		// TODO In the case of a failing writing query, we may hold both locks:
```

Comment:
In case of a writer, `Graph_ReleaseLock` is the function to call to release the read/write lock,
but all values to be written should be pre-computed prior to acquiring the write lock.

Reply:
Yeah, I think we just need to block the unhandled MERGE queries as previously discussed and then this will be an impossible case.